src/ucp/wireup/wireup_cm.c

```diff
@@ -765,23 +760,44 @@ ucp_ep_cm_server_create_connected(ucp_worker_h worker, unsigned ep_init_flags,
                                           ep_init_flags,
                                           "conn_request on uct_listener", &ep);
     if (status != UCS_OK) {
-        return status;
+        ucs_warn("server ep %p failed to connect to worker address on device %s, tl_bitmap 0x%zx, status %s",
+                 ep, conn_request->dev_name, tl_bitmap,
+                 ucs_status_string(status));
+        uct_listener_reject(conn_request->uct.listener, conn_request->uct_req);
```

Comment:
were the rejects in the error flows done before or added in this PR?
can they be handled in a goto label?

Reply:
moved to here from `ucp_listener_conn_request_progress` because the ep can be destroyed and it's too late for reject.
it can't be handled in single goto label, IMHO several err labels look uglier for this function.

Reply:
what is wrong with several error labels?

Reply:
in this func we need to reject, destroy_ep or both. this leads to code duplication in labels or multiple "goto"s